packages/eas-cli/src/update/configure.ts

```diff
@@ -0,0 +1,294 @@
+  runtimeVersionA: NonNullable<ExpoConfig['runtimeVersion']>,
+  runtimeVersionB: NonNullable<ExpoConfig['runtimeVersion']>
+): boolean {
+  if (typeof runtimeVersionA === 'string' && typeof runtimeVersionB === 'string') {
+    return runtimeVersionA === runtimeVersionB;
+  } else if (typeof runtimeVersionA === 'object' && typeof runtimeVersionB === 'object') {
+    return runtimeVersionA.policy === runtimeVersionB.policy;
+  } else {
+    return false;
+  }
+}
+
+/**
+ * Make sure the `app.json` is configured to use EAS Updates.
+ * This does a couple of things:
+ *   - Ensure update URL is set to the project EAS endpoint
+ *   - Ensure runtimeVersion is defined for both or individual platforms
+ *   - Output the changes made, or the changes required to make manually
+ */
+export async function ensureEASUpdatesIsConfiguredInExpoConfigAsync({
+  exp,
+  projectId,
+  projectDir,
+  platform,
+  workflows,
+}: {
+  exp: ExpoConfig;
+  projectId: string;
+  projectDir: string;
+  platform: RequestedPlatform;
+  workflows: Record<Platform, Workflow>;
+}): Promise<{ projectChanged: boolean; exp: ExpoConfig }> {
+  const modifyConfig: Partial<ExpoConfig> = {};
+
+  if (exp.updates?.url !== getEASUpdateURL(projectId)) {
+    modifyConfig.updates = { url: getEASUpdateURL(projectId) };
+  }
+
+  let androidRuntimeVersion = exp.android?.runtimeVersion ?? exp.runtimeVersion;
+  let iosRuntimeVersion = exp.ios?.runtimeVersion ?? exp.runtimeVersion;
+
+  if (
+    (['all', 'android'].includes(platform) && !androidRuntimeVersion) ||
+    (['all', 'ios'].includes(platform) && !iosRuntimeVersion)
+  ) {
+    androidRuntimeVersion = androidRuntimeVersion ?? getDefaultRuntimeVersion(workflows.android);
+    iosRuntimeVersion = iosRuntimeVersion ?? getDefaultRuntimeVersion(workflows.ios);
+
+    if (platform === 'all' && isRuntimeEqual(androidRuntimeVersion, iosRuntimeVersion)) {
+      modifyConfig.runtimeVersion = androidRuntimeVersion;
+    } else {
+      if (['all', 'android'].includes(platform)) {
+        modifyConfig.runtimeVersion = undefined;
+        modifyConfig.android = { runtimeVersion: androidRuntimeVersion };
+      }
+      if (['all', 'ios'].includes(platform)) {
+        modifyConfig.runtimeVersion = undefined;
+        modifyConfig.ios = { runtimeVersion: iosRuntimeVersion };
+      }
+    }
+  }
+
+  if (Object.keys(modifyConfig).length === 0) {
+    return { exp, projectChanged: false };
+  }
+
+  // NOTE(cedric): might be better with a mergeDeep method, or handle in `modifyConfigAsync`
+  const mergedExp = {
+    ...exp,
+    runtimeVersion: modifyConfig.runtimeVersion ?? exp.runtimeVersion,
+    updates: { ...exp.updates, url: modifyConfig.updates?.url ?? exp.updates?.url },
+    android: {
+      ...exp.android,
+      runtimeVersion: modifyConfig.android?.runtimeVersion ?? exp.android?.runtimeVersion,
+    },
+    ios: {
+      ...exp.ios,
+      runtimeVersion: modifyConfig.ios?.runtimeVersion ?? exp.ios?.runtimeVersion,
+    },
```

Comment:
```suggestion
    runtimeVersion: modifyConfig.runtimeVersion ?? exp.runtimeVersion,
    updates: { ...exp.updates, ...modifyConfig.updates },
    android: {
      ...exp.android,
      ...modifyConfig.android
    },
    ios: {
      ...exp.ios,
      ...modifyConfig.ios
    },
```
This should work better